smarts/core/vehicle.py

```diff
@@ -276,15 +276,15 @@ def bounding_box(self):
         self._assert_initialized()
         # Assuming the position is the centre,
         # calculate the corner coordinates of the bounding_box
-        half_length = self.length / 2
-        half_width = self.width / 2
+        origin = self.position[:2]
+        radius = numpy.array([self.width / 2, self.length / 2])
         corners = [(-1, 1), (1, 1), (1, -1), (-1, -1)]
+        heading = self.heading
         return [
             rotate_around_point(
-                point=self.position[:2]
-                + numpy.array(corner) * numpy.array([half_width, half_length]),
-                radians=self.heading,
-                origin=self.position[:2],
+                point=origin + numpy.array(corner) * radius,
+                radians=heading,
+                origin=origin,
```

Comment:
This is another unrelated and minor optimization that came out of the flamegraph analysis that I didn't back out again.  The `position` and `heading` properties are not trivial (they involve computing stuff), so no need to do them multiple times due to the list comprehension loop.